### ISSUE TITLE
MemoryError: std::bad_alloc when having too many axes in boost histogram

### ISSUE THREAD

**Issue by tyjyang** [WMass/WRemnants#65]
This error can be reproduced by setting the `nominal_axes` to `[axis_massWgen, axis_absYVgen, axis_ptVgen, axis_chargeWgen, axis_l_eta_gen, axis_l_pt_gen]` in `scripts/histmakers/w_z_gen_dists.py`. The script has a flag `--singleLeptonHists` for this setup.

This is not caused by errors in the individual axes, as `nominal_axes` can be set to either `[axis_massWgen, axis_absYVgen, axis_ptVgen, axis_chargeWgen]` or `[axis_massWgen, axis_l_eta_gen, axis_l_pt_gen]` and it runs fine.

**Comment by tyjyang** [comment]
200*17*120*2*48*29 = 1.13 billion bins

**Comment by bendavid** [comment]
Right so the problem is likely the "helicity_moments_scale" histogram, where you multiply the number of bins by an addition 9*9

You also missed the underflow/overflow bins for some of the axes, so this would be
200*19*123*2*50*31*81 = 117 billion bins -> 1.9TB which is indeed more than the amount of memory on the machine :)

Now I don't think "helicity_moments_scale" makes sense in this configuration (ie there is no reason to bin that one in terms of the single lepton pt and eta, and it's not even well defined to do so, so I guess you can either remove it in this configuration, or use the reduced set of axes for this one)

**Comment by tyjyang** [comment]
Indeed, so I guess we should modify the behavior of `--singleLeptonHists` so it doesn't add more axes to the the `helicity_moments_scale` histogram. Thanks Josh!
